lib/src/hydrated_bloc_storage.dart

```diff
@@ -20,69 +22,171 @@ abstract class HydratedStorage {
+  /// final byteskey = sha256.convert(utf8.encode(pass)).bytes;
+  /// return HiveAesCipher(byteskey);
+  /// ```
   static Future<HydratedBlocStorage> getInstance({
     Directory storageDirectory,
+    HiveCipher encryptionCipher,
   }) {
     return _lock.synchronized(() async {
+      if (_instance != null) {
+        return _instance;
+      }
+
       final directory = storageDirectory ?? await getTemporaryDirectory();
-      final file = File('${directory.path}/.hydrated_bloc.json');
-      var storage = <String, dynamic>{};
-
-      if (await file.exists()) {
-        try {
-          storage =
-              json.decode(await file.readAsString()) as Map<String, dynamic>;
-        } on dynamic catch (_) {
-          await file.delete();
+      if (!kIsWeb) {
+        Hive.init(directory.path);
+      }
+
+      final box = await Hive.openBox(
+        'niagara+',
+        encryptionCipher: encryptionCipher,
+      );
+
+      final singlet = await CellSinglet.instance(
+        directory,
+        (file) => StringCell(file),
+      );
+
+      final tokens = singlet.tokens;
+      if (tokens.isNotEmpty) {
+        for (final token in tokens) {
+          try {
+            final string = await singlet.read(token);
+            final object = json.decode(string);
+            await box.put(token, object);
+          } on dynamic catch (_) {}
         }
+        await singlet.clear();
       }
 
-      return HydratedBlocStorage._(storage, file);
+      return _instance = HydratedBlocStorage._(box);
     });
   }
 
-  HydratedBlocStorage._(this._storage, this._file);
+  static final _lock = Lock();
+  static HydratedStorage _instance;
+
+  HydratedBlocStorage._(this._box);
+  final Box _box;
 
   @override
   dynamic read(String key) {
-    return _storage[key];
+    if (_box.isOpen) {
+      return _box.get(key);
+    } else {
+      return null;
+    }
   }
 
   @override
   Future<void> write(String key, dynamic value) {
-    return _lock.synchronized(() {
-      _storage[key] = value;
-      return _file.writeAsString(json.encode(_storage));
-    });
+    if (_box.isOpen) {
+      return _lock.synchronized(() => _box.put(key, value));
+    } else {
+      return null;
+    }
   }
 
   @override
   Future<void> delete(String key) {
-    return _lock.synchronized(() {
-      _storage[key] = null;
-      return _file.writeAsString(json.encode(_storage));
-    });
+    if (_box.isOpen) {
+      return _lock.synchronized(() => _box.delete(key));
+    } else {
+      return null;
+    }
   }
 
   @override
   Future<void> clear() {
-    return _lock.synchronized(
-      () async {
-        _storage.clear();
-        if (await _file.exists()) {
-          await _file.delete();
-        }
-      },
-    );
+    if (_box.isOpen) {
+      _instance = null;
+      return _lock.synchronized(_box.deleteFromDisk);
+    } else {
+      return null;
+    }
   }
 }
+
+/// `CellSinglet` is storage of one file.
+class CellSinglet {
```

Comment:
do you still think there's value in maintaining a notion of a CellSinglet/Cell etc if we're only supporting single file for backward compat? I kind of feel it would be simpler to remove the abstract because it's a bit hard to follow in my opinion...could just be because I'm tired though 😛 

Let me know what you think 👍

Reply:
Yep, I was like stripping code I already had. Good idea to boil it down even more👍